include/circt/Dialect/SV/SVInOutOps.td

```diff
@@ -47,40 +47,59 @@ def WireOp : SVOp<"wire", [DeclareOpInterfaceMethods<OpAsmOpInterface>,
     }]>
   ];
 
-  let assemblyFormat = "custom<ImplicitSSAName>(attr-dict) `:` type($result)";
+  let assemblyFormat = [{ (`sym` $sym_name^)? custom<ImplicitSSAName>(attr-dict) 
+                          `:` type($result) }];
   let hasCanonicalizeMethod = true;
 
   let extraClassDeclaration = [{
     Type getElementType() {
       return result().getType().cast<InOutType>().getElementType();
     }
+    
+    //===------------------------------------------------------------------===//
+    // SymbolOpInterface Methods
+    //===------------------------------------------------------------------===//
+
+    /// An WireOp may optionally define a symbol.
+    bool isOptionalSymbol() { return true; }
+
   }];
   let verifier = [{ return ::verifyWireOp(*this); }];
 }
 
-def RegOp : SVOp<"reg", [DeclareOpInterfaceMethods<OpAsmOpInterface>]> {
+def RegOp : SVOp<"reg", [DeclareOpInterfaceMethods<OpAsmOpInterface>,
+                         NonProceduralOp, Symbol]> {
```

Comment:
What does the addition of `NonProceduralOp` do? I see this was already on `RegOp`...

Reply:
Adding `NonProceduralOp` is the right thing to do, but I'd recommend putting it into its own patch.  This forbids them from being in a procedural if/always (etc)